s3torchconnector/tst/e2e/test_e2e_s3_lightning_checkpoint.py

```diff
@@ -0,0 +1,102 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+import torch
+import lightning as L
+from pathlib import Path
+from lightning.pytorch.demos import WikiText2
+from torch.utils.data import DataLoader
+
+from s3torchconnector import S3Checkpoint
+from s3torchconnector.lightning import S3LightningCheckpoint
+
+from models.net import Net
+from models.lightning_transformer import LightningTransformer
+
+
+def test_save_and_load_checkpoint(checkpoint_directory):
+    tensor = torch.rand(3, 10, 10)
+    s3_lightning_checkpoint = S3LightningCheckpoint(region=checkpoint_directory.region)
+    checkpoint_name = "lightning_checkpoint.ckpt"
+    s3_uri = f"{checkpoint_directory.s3_uri}/{checkpoint_name}"
+    s3_lightning_checkpoint.save_checkpoint(tensor, s3_uri)
+    loaded_checkpoint = s3_lightning_checkpoint.load_checkpoint(s3_uri)
+    assert torch.equal(tensor, loaded_checkpoint)
+
+
+def test_load_compatibility_with_s3_checkpoint(checkpoint_directory):
+    tensor = torch.rand(3, 10, 10)
+    checkpoint_name = "general_checkpoint.pt"
+    checkpoint = S3Checkpoint(region=checkpoint_directory.region)
+    s3_uri = f"{checkpoint_directory.s3_uri}/{checkpoint_name}"
+    with checkpoint.writer(s3_uri) as writer:
+        torch.save(tensor, writer)
+    lightning_checkpoint = S3LightningCheckpoint(region=checkpoint_directory.region)
+    loaded_checkpoint = lightning_checkpoint.load_checkpoint(s3_uri)
+    assert torch.equal(tensor, loaded_checkpoint)
+
+
+def test_save_compatibility_with_s3_checkpoint(checkpoint_directory):
+    tensor = torch.rand(3, 10, 10)
+    checkpoint_name = "lightning_checkpoint.pt"
+    lightning_checkpoint = S3LightningCheckpoint(region=checkpoint_directory.region)
+    s3_uri = f"{checkpoint_directory.s3_uri}/{checkpoint_name}"
+    lightning_checkpoint.save_checkpoint(tensor, s3_uri)
+    checkpoint = S3Checkpoint(region=checkpoint_directory.region)
+    loaded_checkpoint = torch.load(checkpoint.reader(s3_uri))
+    assert torch.equal(tensor, loaded_checkpoint)
+
+
+def test_load_trained_checkpoint(checkpoint_directory):
+    dataset = WikiText2(data_dir=Path("/tmp/data"))
+    dataloader = DataLoader(dataset)
+    model = LightningTransformer(vocab_size=dataset.vocab_size)
+    trainer = L.Trainer(fast_dev_run=3)
+    trainer.fit(model=model, train_dataloaders=dataloader)
+    checkpoint_name = "lightning_module_training_checkpoint.pt"
+    s3_uri = f"{checkpoint_directory.s3_uri}/{checkpoint_name}"
+    trainer.save_checkpoint(s3_uri)
+    lightning_checkpoint = S3LightningCheckpoint(region=checkpoint_directory.region)
+    loaded_model = lightning_checkpoint.load_checkpoint(s3_uri)
+    for model_item, loaded_item in zip(
```

Comment:
Do you think `(model_key, model_value), (loaded_key, loaded_item) in ...` would look better, and then you'd not have to do the index lookups later